src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java

```diff
@@ -297,6 +297,23 @@ private void testAjaxPictureUrl() {
         return loginAdminToPage(profileUrl, typeOfPage);
     }
 
+    private <T extends AppPage> T loginAsInstructorToPage(String instructorId, AppUrl url, Class<T> typeOfPage) {
```

Comment:
This method should be moved up to `LoginPage` so it can be reused by other tests in future.

Reply:
You mean move to `BaseActionTest` right? This function is only used in testing

Reply:
I meant `BaseUiTestCase`. Sorry I was looking at the wrong file

Reply:
I see. When writing this function, I was imitating the `BaseUiTestCase#loginAdminToPage` which this function replaces. I think, this function should be moved there, because `LoginPage` seems to only lower level functions like `loginAsInstructor' (fill in email address, click login) whereas this function also logout, navigate to new URL, then login.

Reply:
Oh yes; I did mean `BaseUiTestCase`; my bad 😅